### ISSUE TITLE
CK Editor: Browser - Taking a long time to load [Performance Issue]

### ISSUE THREAD

**Issue by mikebigun** [DNN-Connect/CKEditorProvider#121]
### Problem description
Customer is facing with a huge performance issues when loading folder hierarchy on browsing server files from CKEditor.

### Steps to reproduce
1. Add around 20,000 files/ 2,000 folders to your Portal/0 folder 
2. Synchronize the file System in DNN by going to Content > Assets 
3. Go to any HTML Module 
4. Open the advanced editor. 
5. Click on the Link Button 
6. On the Popup click on `Browse Server`

### Current behavior
With the following number of folder/files it takes more than a minute to wait until all data is loaded to the browser page. 
· 183,033 files 
· 44889 Folders 
· 59.2GB

### Expected behavior 
Customer expects to get lists loaded during the 20 sec or less. 

### Screenshots

![image](https://user-images.githubusercontent.com/22524011/61549619-04ff9b80-aa40-11e9-8231-20b5c1b9ab99.png)

**Comment by mikebigun** [comment]
Dear approver's, I already found a few gaps that requires our attention:

1. Clicking on a folder with a huge amount of files takes very long time. Doing performance analysis I found sql queries are super fast but 80% of a time we spend to build files URL's:
https://github.com/DNN-Connect/CKEditorProvider/blob/072ed60450516760c3fa98ba4633dc6793ec6b2c/Browser/Browser.aspx.cs#L302
In a loop of a 100K files, this method breaks the process. 

2. On initial page loading, it creates folder hierarchy (tested with around 10K folders). The problem is here:
https://github.com/DNN-Connect/CKEditorProvider/blob/072ed60450516760c3fa98ba4633dc6793ec6b2c/Browser/Browser.aspx.cs#L1713

Checking permission for every folder invokes a separate call to database. Most probably we have to create another API to load permissions in one run and then check one by one using local collections. 

That means on top of minor changes inside CKEditor, we have to refactor DNN.Platform. 
Currently I see editor is referencing DNN.Platform 7.4.0 which is pretty old one.
https://github.com/DNN-Connect/CKEditorProvider/blob/072ed60450516760c3fa98ba4633dc6793ec6b2c/packages.config#L1-L4

New fixes will require DNN to be updated here. 

Is there chance we can do that? Any blockers?

@ohine @daguiler
